Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -285,16 +311,41 @@ void OpenALStream::SoundLoop()
 				{
 					alBufferData(uiBufferTemp[iBuffersFilled], AL_FORMAT_51CHN32, dpl2, nSamples * FRAME_SURROUND_FLOAT, ulFrequency);
 				}
+				else if (fixed32_capable)
+				{
+					int surround_int32[OAL_MAX_SAMPLES * SURROUND_CHANNELS * OAL_MAX_BUFFERS];
+
+					for (u32 i = 0; i < nSamples * SURROUND_CHANNELS; ++i)
+					{
+						// For some reason the DPL2 decoder outputs samples bigger than 1. Most are close to 2.5 and some go up to 8. How to deal with this? Hard clamping here, we need to fix the decoder or implement a limiter.
+						if (dpl2[i] >= 1)
+							surround_int32[i] = INT_MAX;
+						else if (dpl2[i] <= -1)
+							surround_int32[i] = INT_MIN;
+						else
+							surround_int32[i] = (int)((float)dpl2[i] * (1 << 31));
+					}
+
+					alBufferData(uiBufferTemp[iBuffersFilled], AL_FORMAT_51CHN32, surround_int32, nSamples * FRAME_SURROUND_INT32, ulFrequency);
+				}
 				else
 				{
 					short surround_short[OAL_MAX_SAMPLES * SURROUND_CHANNELS * OAL_MAX_BUFFERS];
 					for (u32 i = 0; i < nSamples * SURROUND_CHANNELS; ++i)
-						surround_short[i] = (short)((float)dpl2[i] * (1 << 15));
+					{
+						// For some reason the DPL2 decoder outputs samples bigger than 1. Most are close to 2.5 and some go up to 8. How to deal with this? Hard clamping here, we need to fix the decoder or implement a limiter.
+						if (dpl2[i] >= 1)
+							surround_short[i] = SHRT_MAX;
+						else if (dpl2[i] <= -1)
+							surround_short[i] = SHRT_MIN;
+						else
+							surround_short[i] = (int)((float)dpl2[i] * (1 << 15));
```

Comment:
Here is another kind of issue. There are values too close to 1.0, but still smaller 1.0. For example 1.0 - 1.0 / (1<<23) can be represented as a float. So you'll end with 1.0 * (1<<15), which results in -(1<<15) when casted to short.